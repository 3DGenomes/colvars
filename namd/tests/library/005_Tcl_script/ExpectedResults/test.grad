# 1
#         12       0.2       100  0

 12.1  69.9821
 12.3  -526.451
 12.5  -516.771
 12.7  -506.023
 12.9  -488.141
 13.1  -455.017
 13.3  -412.329
 13.5  -359.724
 13.7  -297.684
 13.9  -212.126
 14.1  -97.863
 14.3  -13.8637
 14.5  0
 14.7  0
 14.9  0
 15.1  0
 15.3  0
 15.5  0
 15.7  0
 15.9  0
 16.1  0
 16.3  0
 16.5  0
 16.7  0
 16.9  0
 17.1  0
 17.3  0
 17.5  0
 17.7  0
 17.9  0
 18.1  0
 18.3  0
 18.5  0
 18.7  0
 18.9  0
 19.1  0
 19.3  0
 19.5  0
 19.7  0
 19.9  0
 20.1  0
 20.3  0
 20.5  0
 20.7  0
 20.9  0
 21.1  0
 21.3  0
 21.5  0
 21.7  0
 21.9  0
 22.1  0
 22.3  0
 22.5  0
 22.7  0
 22.9  0
 23.1  0
 23.3  0
 23.5  0
 23.7  0
 23.9  0
 24.1  0
 24.3  0
 24.5  0
 24.7  0
 24.9  0
 25.1  0
 25.3  0
 25.5  0
 25.7  0
 25.9  0
 26.1  0
 26.3  0
 26.5  0
 26.7  0
 26.9  0
 27.1  0
 27.3  0
 27.5  0
 27.7  0
 27.9  0
 28.1  0
 28.3  0
 28.5  0
 28.7  0
 28.9  0
 29.1  0
 29.3  0
 29.5  0
 29.7  0
 29.9  0
 30.1  0
 30.3  0
 30.5  0
 30.7  0
 30.9  0
 31.1  0
 31.3  0
 31.5  0
 31.7  0
 31.9  0
