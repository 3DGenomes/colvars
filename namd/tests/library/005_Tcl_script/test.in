colvarsTrajFrequency    2


colvar {
    name d

    lowerBoundary 12
    upperBoundary 32
    width 0.2

    distance {
        group1 { atomnumbers 4 }
        group2 { atomnumbers 99 }
    }
}


abf {
    colvars        d
    fullSamples    0
    outputFreq     60
}


harmonic {
    colvars d
    centers 12
    forceConstant 10

    outputEnergy
}
