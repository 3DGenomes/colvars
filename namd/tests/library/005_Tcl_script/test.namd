# -*- tcl -*-

# FILENAMES
set outName             [file rootname [file tail [info script]]]
#set inFileNum           [expr [scan [string range $outName end-1 end] "%d"] - 1]
#set inName              [format "%s%02u" [string range $outName 0 end-2] $inFileNum]

# FILES

#outputname              test01
#DCDfile                 test01.dcd

# MD SETUP
timestep          0.5
numSteps		  20

# Interactive molecular dynamics
#IMDon               on
#IMDport             3000
#IMDfreq             20
#IMDwait             on

colvars			  on
colvarsConfig	  test.in
#colvarsInput  $inName.colvars.state
#colvarsInput  test.colvars.state

#tclForces on
#tclForcesScript {
#  addatom 12
#  proc calcforces {} {
#    addforce 12 { 10 0 0 }
#  }
#}

# INPUT
structure               ../Common/da.psf 
#parameters              missing.par
parameters              ../Common/par_all22_prot.inp
paraTypeCharmm          on
coordinates             ../Common/da.min.pdb

temperature 250
#bincoordinates       $inName.coor
#binvelocities        $inName.vel

# OUTPUT

outputname              ${outName}
outputenergies          100
outputtiming            2000
binaryoutput            no
# binaryrestart           yes

#XSTFreq                 1000

# DCD TRAJECTORY

DCDfile                 ${outName}.dcd
DCDfreq                 2000

seed		87654321

# CONSTANT-T
langevin                on
langevinTemp		    300.0
langevinDamping         10.0
langevinHydrogen	    on

# SPACE PARTITIONING
splitpatch              hydrogen
hgroupcutoff            2.8
stepspercycle           20
margin                  1.0

# CUT-OFFS
switching               on
switchdist              9.0
cutoff                  11.0
pairlistdist            12.0

# RESPA 
fullElectFrequency      1
nonbondedFreq           1

# 1-4 NON-BONDED
exclude                 scaled1-4
1-4scaling              1.0

# COM
commotion               no

# SHAKE
rigidBonds              none

proc runcmd { cmd } {
  print "Running $cmd"
  print [eval $cmd]
}

run 0

foreach c [colvars list] { print "$c = [colvars colvar $c value]"}        
foreach b [colvars list biases] { print "E($b) = [colvars bias $b energy]"}

runcmd "colvars update"

foreach c [colvars list] { print "$c = [colvars colvar $c value]"}        
foreach b [colvars list biases] { print "E($b) = [colvars bias $b energy]"}

run 20

foreach c [colvars list] { print "$c = [colvars colvar $c value]"}        
foreach b [colvars list biases] { print "E($b) = [colvars bias $b energy]"}

# colvars config "colvar {
#   name d2
#   distance {
#     group1 { atomnumbers 4 }
#     group2 { atomnumbers 99 }
#   }
# }"
# runcmd "colvars colvar d2 value"
# runcmd "colvars update"
# runcmd "colvars colvar d2 value"
# run 20
# runcmd "colvars colvar d2 value"
# runcmd "colvars colvar d2 delete"

colvars config "harmonic {
  name harm2
  colvars d
  centers 13
  forceConstant 1.
  outputEnergy yes
}"
run 20 

runcmd "colvars bias harmonic1 delete"

runcmd "colvars bias harm2 energy"
runcmd "colvars bias harm2 update"
run 20 
runcmd "colvars bias harm2 energy"
runcmd "colvars bias harm2 delete"

colvars reset
# reinitialize just for the fun of it ;-)
colvars configfile 	  test.in
