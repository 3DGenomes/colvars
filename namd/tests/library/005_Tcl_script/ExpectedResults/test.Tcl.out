TCL: d = 0.00000000000000e+00
TCL: E(abf1) = 1.69499e+166
TCL: E(harmonic1) = 5.04759e+223
TCL: Running cv update
TCL: 
TCL: d = 0.00000000000000e+00
TCL: E(abf1) = 1.69499e+166
TCL: E(harmonic1) = 18000
TCL: Original numsteps 20 will be ignored.
TCL: Running for 20 steps
TCL: d = 1.30566036331419e+01
TCL: E(abf1) = 1.69499e+166
TCL: E(harmonic1) = 139.551
TCL: Running cv colvar d2 value
TCL: 0.00000000000000e+00
TCL: Running cv update
TCL: 
TCL: Running cv colvar d2 value
TCL: 1.30566036331419e+01
TCL: Running for 20 steps
TCL: Running cv colvar d2 value
TCL: 1.23856683573848e+01
TCL: Running cv bias harm2 energy
TCL: 0
TCL: Running cv bias harm2 update
TCL: 3.41736
TCL: Running cv bias harm2 energy
TCL: 3.41736
TCL: Running cv bias harm2 delete
TCL: 
TCL: Running cv colvar d2 delete
TCL: 
